File: Aran_Aim_test/Geometries/Ellipsoid.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Aran.Package;

namespace Aran.Geometries.SpatialReferences
{
	public class Ellipsoid : IPackable
	{
		public SpatialReferenceGeoType SrGeoType{get;set;}
        public double SemiMajorAxis { get; set; }
        public double Flattening { get; set; }
        public bool IsValid { get; set; }

		public Ellipsoid()
		{
			SrGeoType = SpatialReferenceGeoType.srgtWGS1984;
			SemiMajorAxis = 0;
			Flattening = 0;
			IsValid = false;
		}


        public void Pack (PackageWriter writer)
        {
            writer.PutEnum<SpatialReferenceGeoType> (SrGeoType);
            writer.PutDouble (SemiMajorAxis);
            writer.PutDouble (Flattening);
            writer.PutBool (IsValid);
        }

        public void Unpack (PackageReader reader)
        {
            SrGeoType = reader.GetEnum<SpatialReferenceGeoType> ();
            SemiMajorAxis = reader.GetDouble ();
            Flattening = reader.GetDouble ();
            IsValid = reader.GetBool ();
        }
    }
}

File: Aran/RNAV/Libs/Panda/GeometryOperatorsContract/cpp/src/Ellipsoid.cpp
#include "../include/Ellipsoid.h"

namespace Panda
{
	Ellipsoid::Ellipsoid():
		srGeoType(0),
		semiMajorAxis(0),
		flattening(0),
		isValid(false)
	{
	}

	Ellipsoid::~Ellipsoid()
	{
	}

	void Ellipsoid::pack (Handle handle) const throw (Registry::Exception)
	{
		Registry::putInt32 (handle, srGeoType);
		Registry::putDouble (handle, semiMajorAxis);
		Registry::putDouble (handle, flattening);
		Registry::putBool (handle, isValid);
	}

	void Ellipsoid::unpack (Handle handle) throw (Registry::Exception)
	{
		Registry::getInt32 (handle, srGeoType);
		Registry::getDouble (handle, semiMajorAxis);
		Registry::getDouble (handle, flattening);
		Registry::getBool (handle, isValid);
	}
}